### ISSUE TITLE
Why browser version needs to be compiled separately?

### ISSUE THREAD

**Issue by tscislo** [raml-org/raml-js-parser-2#521]
It's a little overhead to compile browser version separately each time base npm code changes. Why can't it be provided in the npm package out of the box?

Is the bowser version available somewhere as a standalone npm package?

**Comment by ddenisenko** [comment]
Browser version is attached to each release. It was removed from NPM package due to user complains regarding browser version increasing NPM package size.

**Comment by tscislo** [comment]
The problem is NPM package. I'd like to use it from NPM, but in case of any update I need to compile that manually myself which is an overhead. Can you please provide it in a separate NPM module which will be maintained and updated by you?

**Comment by tscislo** [comment]
The other thing is that browser version is not minified, and when I forced minification it trowed me some exceptions when I was trying to use it.

**Comment by ddenisenko** [comment]
Hm. If we decide to provide an automatic way of upgrading the browser version of the package, something like bower looks more native for this kind of apps than npm.

**Comment by tscislo** [comment]
Bower is a very obsolate package manager. In fact you can manage and maitain all your frontend dependencies in NPM now. They are easilty consumable by gulp, grunt, webpack. As I mentioned at the very beginning it's very problematic for me to rebuild frontend version each time npm version changes. It should be a ready to go browser version provided in a separate npm package.

**Comment by ddenisenko** [comment]
Btw, what would you like to see inside those NPM packages for browser version? Single js file, obviously, but what else? Anything special in package.json?

**Comment by ddenisenko** [comment]
I am splitting this ticket into bower and npm parts: https://github.com/raml-org/raml-js-parser-2/issues/536 and https://github.com/raml-org/raml-js-parser-2/issues/537 to avoid discussion crowding. Please provide any details regarding how you want npm to look in the respective ticket.

This one will be closed as soon as both parts are implemented.

**Comment by tscislo** [comment]
Cool thanks!

**Comment by TomaszScisloGS** [comment]
Any ideas why with the brwser version when I try to import it with webpack I get the following errors:

```
WARNING in C:/Developer/Projects/api-store/apistore-client/~/raml-1-parser-browser/index.js
Module not found: Error: Cannot resolve module 'RAML.JsonValidation' in C:\Developer\Projects\api-store\apistore-client\node_modules\raml-1-parser-browser
 @ C:/Developer/Projects/api-store/apistore-client/~/raml-1-parser-browser/index.js 1:140-170

WARNING in C:/Developer/Projects/api-store/apistore-client/~/raml-1-parser-browser/index.js
Module not found: Error: Cannot resolve module 'RAML.XmlValidation' in C:\Developer\Projects\api-store\apistore-client\node_modules\raml-1-parser-browser
 @ C:/Developer/Projects/api-store/apistore-client/~/raml-1-parser-browser/index.js 1:207-236
```

**Comment by dreamflyer** [comment]
It because of RAML.JsonValidation && RAML. XmlValidation are external dependencies which are optional. You can simply specify in your webpack config:
`externals: [
            {
                "RAML.JsonValidation": true,
                "RAML.XmlValidation": true
            }
        ]`

**Comment by dreamflyer** [comment]
will be published in v1.1.9.
